klient/wait/conditions/conditions.go

```diff
@@ -58,6 +60,120 @@ func (c *Condition) ResourceScaled(obj k8s.Object, scaleFetcher func(object k8s.
 	}
 }
 
+// ResourceMatch is a helper function used to check if the resource under question has met a pre-defined state. This can
+// be leveraged for checking fields on a resource that may not be immediately present upon creation.
+func (c *Condition) ResourceMatch(obj k8s.Object, matchFetcher func(object k8s.Object) bool) apimachinerywait.ConditionFunc {
+	return func() (done bool, err error) {
+		if err := c.resources.Get(context.TODO(), obj.GetName(), obj.GetNamespace(), obj); err != nil {
+			return false, nil
+		}
+		return matchFetcher(obj), nil
+	}
+}
+
+// ResourceListN is a helper function that can be used to check for a minimum number of returned objects in a list. This function
+// accepts list options that can be used to adjust the set of objects queried for in the List resource operation.
+func (c *Condition) ResourceListN(list k8s.ObjectList, n int, listOptions ...resources.ListOption) apimachinerywait.ConditionFunc {
+	return c.ResourceListMatchN(list, n, func(object k8s.Object) bool { return true }, listOptions...)
+}
+
+// ResourceListMatchN is a helper function that can be used to check for a minimum number of returned objects in a list. This function
+// accepts list options and a match function that can be used to adjust the set of objects queried for in the List resource operation.
+func (c *Condition) ResourceListMatchN(list k8s.ObjectList, n int, matchFetcher func(object k8s.Object) bool, listOptions ...resources.ListOption) apimachinerywait.ConditionFunc {
+	return func() (done bool, err error) {
+		if err := c.resources.List(context.TODO(), list, listOptions...); err != nil {
+			return false, nil
+		}
+		var found int
+		metaList, err := meta.ExtractList(list)
+		if err != nil {
+			return false, err
+		}
+		for _, obj := range metaList {
+			if o, ok := obj.(k8s.Object); ok && matchFetcher(o) {
+				found++
+			} else if !ok {
+				return false, fmt.Errorf("unexpected type %T in list, does not satisfy k8s.Object", obj)
```

Comment:
Suggestion: add a context to this error (and other errors like it), something like:

```go
fmt.Errorf("condition: resource match: type %T does not satisfy k8s.Object")
```